source/core/index.ts

```diff
@@ -731,10 +731,13 @@ export default class Request extends Duplex implements RequestEvents<Request> {
 
 			const updatedOptions = new Options(undefined, undefined, this.options);
 
-			const shouldBeGet = statusCode === 303 && updatedOptions.method !== 'GET' && updatedOptions.method !== 'HEAD';
-			if (shouldBeGet || updatedOptions.methodRewriting) {
-				// Server responded with "see other", indicating that the resource exists at another location,
-				// and the client should request it from that location via GET or HEAD.
+			const serverRequestedGet = statusCode === 303 && updatedOptions.method !== 'GET' && updatedOptions.method !== 'HEAD';
+			const canRewrite = statusCode !== 307 && statusCode !== 308;
+			const userRequestedGet = updatedOptions.methodRewriting && canRewrite;
+			if (serverRequestedGet || userRequestedGet) {
+				// Either the server responded with 303 or
+				// the methodRewriting option explicitly requests
+				// a rewrite to GET on non-307/308 responses
```

Comment:
Given the logic in the above, this comment is kinda moot now.

Reply:
agree. Removed.